### PR TITLE
Replace RxSwift for `ManifestLoader`

### PR DESCRIPTION
**Pull request by ezraberch**
Part of #4004

### Short description 📝

Not much to do for this one, since RxSwift isn't used in an asynchronous manner here.

### How to test the changes locally 🧐

Covered by existing tests

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).

### CODE REVIEW

File: Sources/TuistLoader/Loaders/ManifestLoader.swift [hunk 1]
```diff
@@ -227,18 +225,12 @@ public class ManifestLoader: ManifestLoading {
...
 
-        switch result {
-        case let .completed(elements):
-            let output = elements.filter(\.isStandardOutput).map(\.value).reduce(into: Data()) { $0.append($1) }
-            guard let string = String(data: output, encoding: .utf8) else { return output }
-
-            guard let startTokenRange = string.range(of: ManifestLoader.startManifestToken) else { return output }
-            guard let endTokenRange = string.range(of: ManifestLoader.endManifestToken) else { return output }
+            guard let startTokenRange = string.range(of: ManifestLoader.startManifestToken)
+            else { return string.data(using: .utf8)! }
```
Comment (danyf90) [inline Sources/TuistLoader/Loaders/ManifestLoader.swift:232]: For consistency with the line below
```suggestion
            guard let startTokenRange = string.range(of: ManifestLoader.startManifestToken) else { return string.data(using: .utf8)! }
```

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistLoader/Loaders/ManifestLoader.swift (hunk 1 above)
Comment (ezraberch) [reply Sources/TuistLoader/Loaders/ManifestLoader.swift]: That's what I did initially, but SwiftFormat complained:

```
ManifestLoader.swift:231:1: warning: (wrap) Wrap lines that exceed the specified maximum width.
```

If you want, I can instead change the other line to match.

File: Sources/TuistLoader/Loaders/ManifestLoader.swift [hunk 2]
```diff
@@ -1,7 +1,5 @@
 import Foundation
 import ProjectDescription
-import RxBlocking
-import RxSwift
```
Comment (laxmorek) [inline Sources/TuistLoader/Loaders/ManifestLoader.swift:4]: 🎉

**Review by laxmorek — APPROVED** [review]

File: Sources/TuistLoader/Loaders/ManifestLoader.swift [hunk 3]
```diff
@@ -227,18 +225,11 @@ public class ManifestLoader: ManifestLoading {
             at: path
         ) + ["--tuist-dump"]
 
-        let result = System.shared
-            .observable(arguments, verbose: false, environment: environment.manifestLoadingVariables)
-            .toBlocking()
-            .materialize()
+        do {
+            let string = try System.shared.capture(arguments, verbose: false, environment: environment.manifestLoadingVariables)
```
Comment (adellibovi) [inline Sources/TuistLoader/Loaders/ManifestLoader.swift:229]: Worth knowing if there was any reason why before we were using the other method. Is `capture` getting only the standard output as well? or both standard and error output? could you double-check? 😄

**Review by adellibovi — CHANGES_REQUESTED** [review]

**Comment by adellibovi** [comment]
Thanks @ezraberch :)

File: Sources/TuistLoader/Loaders/ManifestLoader.swift (hunk 1 above)
Comment (danyf90) [reply Sources/TuistLoader/Loaders/ManifestLoader.swift]: Sorry, fixed it 🙏

File: Sources/TuistLoader/Loaders/ManifestLoader.swift (hunk 3 above)
Comment (ezraberch) [reply Sources/TuistLoader/Loaders/ManifestLoader.swift:229]: Capture does get the error output and passes it back (as an exception) if the process fails.

There are tests (`test_load_invalidFormat` and `test_load_missingManifest`) which cover this.

On a related note, it seems like 2 of the 4 `ManifestLoaderError` cases don't actually get used. In the case of the code being changed here, the error from the underlying system call is simply rethrown. It seems like `ManifestLoaderError.unexpectedOutput` might be more appropriate, but using that would display less information. Perhaps just remove the unused cases?

File: Sources/TuistLoader/Loaders/ManifestLoader.swift (hunk 1 above)
Comment (ezraberch) [reply Sources/TuistLoader/Loaders/ManifestLoader.swift]: Works for me.

**Review by adellibovi — APPROVED** [review]

**Pull request merged**